app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -49,6 +49,62 @@ class PatientRepository @Inject constructor(
 
   private var ongoingNewPatientEntry: OngoingNewPatientEntry = OngoingNewPatientEntry()
 
+  fun search(name: String): Observable<List<PatientSearchResult>> {
+    return configProvider.flatMapObservable { (isFuzzySearchV2Enabled) ->
+      if (isFuzzySearchV2Enabled) {
+        searchWithoutAgeV2(name = name)
+      } else {
+        searchWithoutAgeV1(name = name)
+      }
+    }
+  }
+
+  @Deprecated(message = "replaced by search v2")
+  private fun searchWithoutAgeV1(name: String): Observable<List<PatientSearchResult>> {
+    val searchableName = nameToSearchableForm(name)
+
+    val substringSearch = database.patientSearchDao()
+        .search(searchableName, PatientStatus.ACTIVE)
+        .toObservable()
+
+    val fuzzySearch = database.fuzzyPatientSearchDao()
+        .searchForPatientsWithNameLike(searchableName)
+        .toObservable()
+
+    return Observables.zip(substringSearch, fuzzySearch, configProvider.toObservable())
+        .map { (results, fuzzyResults, config) ->
+          // Fuzzy search has an internal limit, but the substring search also can return more than
+          // a thousand results which can fail when we sort on the current facility since it
+          // queries the blood pressure table with an IN clause which can cause an exception.
+          (fuzzyResults + results).distinctBy { it.uuid }.take(config.limitOfSearchResults)
+        }
+        .compose(sortByCurrentFacility())
+  }
+
+  private fun searchWithoutAgeV2(name: String): Observable<List<PatientSearchResult>> {
+    return database.patientSearchDao()
+        .nameAndId(PatientStatus.ACTIVE)
+        .toObservable()
+        .switchMapSingle { searchPatientByName.search(name, it) }
+        .zipWith(configProvider.toObservable())
```

Comment:
Why does this have to be a zip? We won't get updates if the database is updated.

Reply:
Duplicate comment?